### PR TITLE
kms: cache results

### PR DESCRIPTION
**Pull request by serprex**
kms should not be polled repeatedly, keep an in-memory cache of decrypted kms env variables

### CODE REVIEW

File: flow/peerdbenv/env.go [hunk 1]
```diff
@@ -86,7 +87,15 @@ func decryptWithKMS(ctx context.Context, data []byte) ([]byte, error) {
 	return decrypted.Plaintext, nil
 }
 
+var KMSCache sync.Map
+
 func GetEnvBase64EncodedBytes(ctx context.Context, name string, defaultValue []byte) ([]byte, error) {
+	if cacheVal, ok := KMSCache.Load(name); ok {
```
Comment (serprex) [inline flow/peerdbenv/env.go:93]: There is a slight race here where two concurrent accesses will run in parallel. This is fine

File: flow/peerdbenv/env.go [hunk 2]
```diff
@@ -86,7 +87,15 @@ func decryptWithKMS(ctx context.Context, data []byte) ([]byte, error) {
 	return decrypted.Plaintext, nil
 }
 
+var KMSCache sync.Map
+
 func GetEnvBase64EncodedBytes(ctx context.Context, name string, defaultValue []byte) ([]byte, error) {
```
Comment (heavycrystal) [inline flow/peerdbenv/env.go:92]: this function is used for fetching Temporal cert and key too
are they also going to be KMS encrypted

also function name should indicate KMS dependency

File: flow/peerdbenv/env.go [hunk 3]
```diff
@@ -86,7 +87,15 @@ func decryptWithKMS(ctx context.Context, data []byte) ([]byte, error) {
 	return decrypted.Plaintext, nil
 }
 
+var KMSCache sync.Map
```
Comment (heavycrystal) [inline flow/peerdbenv/env.go:90]: don't see why this has to be public?

File: flow/peerdbenv/env.go (hunk 2 above)
Comment (serprex) [reply flow/peerdbenv/env.go:92]: yes

**Review by iskakaushik — APPROVED** [review]

**Pull request merged**